src/dataServices/hybridDataService.ts

```diff
@@ -0,0 +1,323 @@
+
+            // link the s3 key to the stripped resource
+            if (!strippedResource.meta) {
+                strippedResource.meta = { source: fileName };
+            } else {
+                strippedResource.meta = { ...strippedResource.meta, source: fileName };
+            }
+
+            try {
+                const [updateResponse, s3UploadResult] = await Promise.all([
+                    this.dbPersistenceService.updateResourceWithOutCheck(resourceType, strippedResource, id, tenantId),
+                    S3ObjectStorageService.uploadObject(
+                        encode(JSON.stringify(resourceClone)),
+                        fileName,
+                        'application/json',
+                    ),
+                ]);
+                resourceClone.meta = updateResponse.resource.meta;
+                assert(s3UploadResult.message, '');
+                return {
+                    success: updateResponse.success,
+                    message: updateResponse.message,
+                    resource: resourceClone,
+                };
+            } catch (e) {
+                await this.dbPersistenceService.deleteResource({ resourceType: request.resourceType, id });
+                throw e;
+            }
+        } else {
+            return this.dbPersistenceService.updateResource(request);
+        }
+    }
+
+    async deleteResource(request: DeleteResourceRequest) {
+        this.assertValidTenancyMode(request.tenantId);
+        const { resourceType, id, tenantId } = request;
+        const itemServiceResponses = await this.dbPersistenceService.readAllResourcesMetaData({
+            resourceType,
+            id,
+            tenantId,
+        });
+        console.log(`Found resources to delete: ${itemServiceResponses.resource.length}`);
+        await Promise.all(
+            itemServiceResponses.resource.map(async (element: any) => {
+                const { versionId, source } = element.meta;
+
+                console.log(`Found versionId to delete: ${versionId}`);
+                console.log(`Found source to delete: ${source}`);
+                if (source) {
+                    const [, deleteResponse] = await Promise.all([
+                        S3ObjectStorageService.deleteObject(source),
+                        this.dbPersistenceService.deleteVersionedResource(id, parseInt(versionId, 10), tenantId),
+                    ]);
+                    console.log(`Deleted version: ${deleteResponse.message}`);
+                    return deleteResponse;
+                }
+                return this.dbPersistenceService.deleteVersionedResource(id, parseInt(versionId, 10), tenantId);
+            }),
+        );
+        return {
+            success: true,
+            message: `Successfully deleted resource Id: ${id}`,
+        };
+    }
+
+    async deleteOldResource(request: DeleteResourceRequest) {
```

Comment:
I had a look around the changes and didn't find a reference for this. Just curious when / how this is used.